corehq/apps/translations/forms.py

```diff
@@ -89,88 +90,67 @@ def __init__(self, domain, *args, **kwargs):
-                                    )
-    # Unfortunately transifex api does not provide a way to pull all possible target languages and
-    # allow us to just add a checkbox instead of selecting a single/multiple target languages at once
-    target_lang = forms.ChoiceField(label=ugettext_lazy("Target Language"),
-                                    choices=([(None, ugettext_lazy('Select Target Language'))] +
+    target_lang = forms.ChoiceField(label=ugettext_lazy("Translated Language"),
+                                    choices=([(None, ugettext_lazy('Select Translated Language'))] +
                                              langcodes.get_all_langs_for_select()),
-                                    help_text=ugettext_lazy("Leave blank to skip"),
                                     required=False,
                                     )
-    action = forms.ChoiceField(label=ugettext_lazy("Action"),
-                               choices=[('push', ugettext_lazy('Push to transifex')),
-                                        ('pull', ugettext_lazy('Pull from transifex')),
-                                        ('delete', ugettext_lazy('Delete resources on transifex'))]
-                               )
-    lock_translations = forms.BooleanField(label=ugettext_lazy("Lock resources"),
-                                           help_text=ugettext_lazy(
-                                               "Lock translations for resources that are being pulled"),
-                                           required=False,
-                                           initial=False)
-    perform_translated_check = forms.BooleanField(label=ugettext_lazy("Check for completion"),
-                                                  help_text=ugettext_lazy(
-                                                      "Check for translation completion before pulling files"),
-                                                  required=False,
-                                                  initial=True)
+    action = forms.CharField(widget=forms.HiddenInput)
+    perform_translated_check = forms.BooleanField(
+        label=ugettext_lazy("Confirm that resources are completely translated before performing request"),
+        required=False,
+        initial=True)
 
     def __init__(self, domain, *args, **kwargs):
         super(AppTranslationsForm, self).__init__(*args, **kwargs)
         self.domain = domain
         self.helper = FormHelper()
         self.helper.form_tag = False
-        self.helper.label_class = 'col-sm-3 col-md-4 col-lg-2'
-        self.helper.field_class = 'col-sm-4 col-md-5 col-lg-3'
+        self.helper.label_class = 'col-sm-4 col-md-4 col-lg-3'
+        self.helper.field_class = 'col-sm-6 col-md-6 col-lg-5'
 
         self.fields['app_id'].choices = tuple((app.id, app.name) for app in get_brief_apps_in_domain(domain))
-        if settings.TRANSIFEX_DETAILS:
+        projects = TransifexProject.objects.filter(domain=domain).all()
```

Comment:
seems like this could be a cached DB accessor

Reply:
I think it's fine to leave this as-is. It's going to be called only when the form is rendered and plus only for the domains that will add this.